adapters/handlers/rest/handlers_authz.go

```diff
@@ -319,28 +318,36 @@ func (h *authZHandlers) getRolesForUser(params authz.GetRolesForUserParams, prin
 		return authz.NewGetRolesForUserBadRequest().WithPayload(errPayloadFromSingleErr(fmt.Errorf("role name is required")))
 	}
 
-	if err := h.authorizer.Authorize(principal, authorization.READ, authorization.Users(params.ID)...); err != nil {
-		return authz.NewGetRolesForUserForbidden().WithPayload(errPayloadFromSingleErr(err))
-	}
-
-	roles, err := h.controller.GetRolesForUser(params.ID)
+	existedRoles, err := h.controller.GetRolesForUser(params.ID)
 	if err != nil {
 		return authz.NewGetRolesForUserInternalServerError().WithPayload(errPayloadFromSingleErr(err))
 	}
 
 	response := []*models.Role{}
 
-	for roleName, policies := range roles {
+	var authErr error
+	for roleName, policies := range existedRoles {
 		perms, err := conv.PoliciesToPermission(policies...)
 		if err != nil {
 			return authz.NewGetRolesInternalServerError().WithPayload(errPayloadFromSingleErr(err))
 		}
+
+		// verify the user has access
+		if err := h.authorizer.Authorize(principal, authorization.READ, authorization.Roles(roleName)...); err != nil {
+			authErr = err
+			continue
+		}
```

Comment:
wouldn't it be easier to just return a forbidded error here? 🤔

Reply:
nope, because there could be many roles and the users has access to 1 of them only

Reply:
so we return everything that the caller has access to and ignore all authErrors unless there is not a single role the caller can access?

Reply:
exactly 

let's the user has many roles 
- RoleA
- RoleB 
 but the user requesting this request has access to only RoleA to read
 that's why we decide on length https://github.com/weaviate/weaviate/pull/6427/files#diff-94644b2fbe6ecbfccdc283fb7d87a3386c6730ddc08423e63554ed48d7ad62a6R347